lib/fiddle/struct.rb

```diff
@@ -152,6 +187,9 @@ def CStructEntity.size(types)
     #
     # See also Fiddle::Pointer.new
     def initialize(addr, types, func = nil)
+      if func && addr.is_a?(Pointer) && addr.free
+        raise ArgumentError, 'free function specified on both underlying struct Pointer and when creating a CStructEntity - who do you want to free this?'
+      end
```

Comment:
Why should we need to accept `Fiddle::Pointer` as `addr`?

Reply:
That's what `addr` is - a pointer - why wouldn't we accept a pointer where the user wants to give us an address? It's currently supported by the API - I didn't add it - I just made it safer here by not allowing two things to think they're responsible for the free operation.

Reply:
I see.